### PR TITLE
Add Key Navigation Filter Menu

### PR DESCRIPTION
**Pull request by tdervieux**
#### New Feature
- The **navigation** menu allows specifying which elements the arrow keys will snap on.
- **Highlight**. When an element is selected, its border becomes green. Now, the thickness of the border is thicker for better readability in mac. The thickness of the border is set by default to 5.
```python
HIGHLIGHT_WIDTH = 5
```

#### Changes
- Refactor how the navigation works. It does not rely on the pyside call *QtWidgets.QGraphicsScene.itemAt* any more but uses the otio methods to navigate from a otio item to another one.
- in *timeline_widget._BaseItem*, the pen is set to pen.setCosmetic(True) so the thickness of the border is not affected by the zoom.
- Ensure that a selected *timeline_widget._BaseItem* is always on the front.
- Minor changes in *timeline_widget.Ruler*. A lambda has been redefined as a function to follow pep8 indications. Also *mouseReleaseEvent* has been subclassed to unselect the ruler so that it does not mess up with the currentSelection.
- Ensure that the ruler is always on the front.

### CODE REVIEW

File: opentimelineview/console.py [hunk 1]
```diff
@@ -190,6 +197,24 @@ def _change_track(self):
...
+
+        def __callback():
+            self._navegation_filter_callback(actions)
+        navegation_menu.triggered[[QtWidgets.QAction]].connect(__callback)
+
+    def _navegation_filter_callback(self, filters):
+        nav_filter = 0
+        for filter in filters:
+            if filter.isChecked():
+                nav_filter += otioViewWidget.timeline_widget.\
```
[the remark this answers is not in the document]
Comment (tdervieux) [reply opentimelineview/console.py]: Good idea. Done.

File: opentimelineview/console.py [hunk 2]
```diff
@@ -131,6 +132,12 @@ def __init__(self, adapter_argument_map, media_linker, *args, **kwargs):
         file_menu.addSeparator()
         file_menu.addAction(exit_action)
 
+        # navegation menu
+        navegation_menu = QtWidgets.QMenu()
+        navegation_menu.setTitle("Navigation")
+        menubar.addMenu(navegation_menu)
```
[the remark this answers is not in the document]
Comment (tdervieux) [reply opentimelineview/console.py]: done.

File: opentimelineview/console.py [hunk 3]
```diff
@@ -29,6 +29,7 @@
 import sys
 import argparse
 import ast
+from collections import OrderedDict
```
[the remark this answers is not in the document]
Comment (tdervieux) [reply opentimelineview/console.py]: done.

[the remark this answers is not in the document]
Comment (tdervieux) [reply opentimelineview/console.py]: I modified timeline_widget.py according to this comment.

**Comment by ssteinbach** [comment]
Thank you @tdervieux!

**Pull request merged**